### PR TITLE
Improve documentation

### LINKED ISSUE #9: Crate docs
The documentation looks a bit... thin right now :slightly_smiling_face: 

2020-08-24-174642_1303x538_scrot

Adding some docs (explaining how the crate is meant to be used and what the functions do) would definitely improve things!

### PR DESCRIPTION
**Pull request by rnestler**
Should fix #9 at least to some extent.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+//! The I2C helpers work with any `embedded_hal::i2c` implementation.
+//!
+//! ```
+//! use embedded_hal_mock::i2c::{Mock as I2cMock, Transaction as I2cTransaction};
+//! use sensirion_i2c::i2c;
+//!
+//! let expectations = [I2cTransaction::write(0x12, vec![0x34, 0x56])];
+//! let mut i2c_mock = I2cMock::new(&expectations);
+//! i2c::write_command(&mut i2c_mock, 0x12, 0x3456);
+//!
```
Comment (dbrgn) [inline src/lib.rs:28]: Nit: Trailing newline

File: src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,33 @@
+//! Library with common functionality for I2C based sensors from Sensirion, based on the
+//! `embedded-hal` traits.
+//!
+//! ## Usage
+//!
+//! ### CRC8 calculation / validation
```
Comment (dbrgn) [inline src/lib.rs:6]: Depending on the markdown dialect, you might need an empty line between the heading and the code block. (I think it's OK for rustdoc though.)

**Review by dbrgn — APPROVED** [review]
Thanks! (I'd still recommend `#![deny(missing_docs)]`, but that's up to you.)

File: src/lib.rs (hunk 1 above)
Comment (rnestler) [reply src/lib.rs]: Fixed :slightly_smiling_face:

File: src/lib.rs (hunk 2 above)
Comment (rnestler) [reply src/lib.rs:6]: Fixed anyways :slightly_smiling_face:

**Review by dbrgn — APPROVED** [review]

**Review by psachs — APPROVED** [review]

**Pull request merged**